### PR TITLE
System.Boolean F# snippets

### PR DESCRIPTION
**Pull request by albert-du**
## Summary

Adds F# snippets for System.Boolean. Revises the article a little to mention the F# examples.

### CODE REVIEW

**Comment by albert-du** [comment]
@dsyme @baronfel could you review?

File: samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/format3.fs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            if arg :? bool && formatProvider = this then
+                let value = arg :?> bool
+                match culture.Name with 
+                | "en-US" -> string arg
+                | "fr-FR" when value -> "vrai"
+                | "fr-FR" -> "faux"
+                | "ru-RU" when value -> "верно"
+                | "ru-RU" -> "неверно"
+                | _ -> string arg
+            else null
```
Comment (baronfel) [inline samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/format3.fs:19]: nitpick: what do you think about this version of the code?

```suggestion
        member this.Format(_, arg, formatProvider) =
            if formatProvider <> this then null
            else
                match arg with
                | :? bool as value -> 
                    match culture.Name with 
                    | "en-US" -> string arg
                    | "fr-FR" when value -> "vrai"
                    | "fr-FR" -> "faux"
                    | "ru-RU" when value -> "верно"
                    | "ru-RU" -> "неверно"
                    | _ -> string arg
                | _ -> null
```

In your original version you had to type-cast the arg twice, but in this one we've introduced a scope where we know `value` must be bool without duplicating work.

Separately, @dsyme should we look at incorporating fantomas into this repo for formatting the samples according to the style guide?

File: samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/tostring2.fs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module tostring2
+
+// <Snippet4>
+let raining = false
+let busLate = true
+
+printfn "It is raining: %s" (if raining then "Yes" else "No")
```
Comment (baronfel) [inline samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/tostring2.fs:7]: I think we're preferring interpolation generally here, so these could be 
```suggestion
printfn $"It is raining: %s{if raining then "Yes" else "No"}" 
```

instead if you wanted.

File: samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.tryparse/fs/tryparseex.fs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+let values = 
+    [ null; String.Empty; "True"; "False"
+      "true"; "false"; "    true    "; "0"
+      "1"; "-1"; "string" ]
+for value in values do
+    match Boolean.TryParse value with
+    | true, flag ->
+        printfn $"'{value}' --> {flag}"
+    | false, _ ->
+        printfn "Unable to parse '%s'." (if value = null then "<null>" else value)
```
Comment (baronfel) [inline samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.tryparse/fs/tryparseex.fs:13]: same here, interpolation everywhere please

File: xml/System/Boolean.xml [hunk 4]
```diff
@@ -108,19 +108,22 @@ A <xref:System.Boolean> instance can have either of two values: `true` or `false
...
 
- The following example illustrates formatting with the <xref:System.Boolean.ToString%2A> method. Note that the example uses the [composite formatting](/dotnet/standard/base-types/composite-formatting) feature, so the <xref:System.Boolean.ToString%2A> method is called implicitly.
+ The following example illustrates formatting with the <xref:System.Boolean.ToString%2A> method. Note that the C# and VB examples use the [composite formatting](/dotnet/standard/base-types/composite-formatting) feature, while the F# example uses [string interpolation](/dotnet/fsharp/language-reference/interpolated-strings) so the <xref:System.Boolean.ToString%2A> method is called implicitly.
```
Comment (baronfel) [inline xml/System/Boolean.xml:111]:
```suggestion
 The following example illustrates formatting with the <xref:System.Boolean.ToString%2A> method. Note that the C# and VB examples use the [composite formatting](/dotnet/standard/base-types/composite-formatting) feature, while the F# example uses [string interpolation](/dotnet/fsharp/language-reference/interpolated-strings). In both cases the <xref:System.Boolean.ToString%2A> method is called implicitly.
```

**Review by baronfel — CHANGES_REQUESTED** [review]
A couple questions, and if you can please use string interpolation everywhere. I love the attention to detail in the editorial text changes, too!

File: samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/tostring2.fs (hunk 2 above)
Comment (albert-du) [reply samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/tostring2.fs:7]: There's a weird bug that doesn't allow for string literals inside interpolated strings. It does work inside triple quoted strings.

This doesn't compile:
```fsharp
printfn $"It is raining: %s{if raining then "Yes" else "No"}"
``` 

This does:
```fsharp
printfn $"""It is raining: %s{if raining then "Yes" else "No"}"""`
```

https://sharplab.io/#v2:DYLgZgzgNALiBOBXAdlAJiA1AHwA7wEtkYxkACAEgCIYALAiMtAewFMJkByGMgd2fgBrMgG8CYMjCStJtVuSoBPdlTKtgEGVWTNVAXyoBYAFD4iJctSo16jFuz4DhYiVMQy68skpVqNWnX1rKiA=

Comment (albert-du) [reply samples/snippets/fsharp/VS_Snippets_CLR_System/system.boolean.structure/fs/tostring2.fs:7]: I'll change it to string interpolation with triple quotes.

**Review by baronfel — APPROVED** [review]
Thanks for making those changes!

:shipit:

**Comment by albert-du** [comment]
@BillWagner

**Review by dsyme — APPROVED** [review]

**Comment by albert-du** [comment]
@BillWagner ready for merge

**Review by dsyme — APPROVED** [review]

**Comment by albert-du** [comment]
@BillWagner Ready for merge

**Pull request merged**